s3/src/main/scala/monix/connect/s3/S3.scala

```diff
@@ -885,6 +885,110 @@ trait S3 { self =>
     } yield s3Object
   }
 
+  /**
+    * Returns the most recently uploaded object in a bucket.
+    *
+    * ==Example==
+    *
+    * {{{
+    * import cats.effect.Resource
+    * import monix.connect.s3.S3
+    * import monix.eval.Task
+    * import software.amazon.awssdk.auth.credentials.{AwsBasicCredentials, StaticCredentialsProvider}
+    * import software.amazon.awssdk.regions.Region
+    * import monix.execution.Scheduler.Implicits.global
+    *
+    * val bucket = "my-bucket"
+    * val prefix = "prefix/to/list/keys"
+    * val s3AccessKey: String = "TESTKEY"
+    * val s3SecretKey: String = "TESTSECRET"
+    * val basicAWSCredentials = AwsBasicCredentials.create(s3AccessKey, s3SecretKey)
+    * val staticCredProvider = StaticCredentialsProvider.create(basicAWSCredentials)
+    *
+    * val s3: Resource[Task, S3] = S3.create(staticCredProvider, Region.AWS_GLOBAL)
+    *
+    * s3.use{s3 => s3.listLatestObject(bucket, prefix = Some(prefix)).toListL}.runToFuture
+    * }}}
+    *
+    * To use this operation in an AWS (IAM) policy, you must have permissions to perform
+    * the `ListBucket` action. The bucket owner has this permission by default and can grant it.
+    *
+    * @param bucket        target S3 bucket name of the object to be downloaded.
+    * @param prefix        limits the response to keys that begin with the specified prefix.
+    * @param requestPayer  confirms that the requester knows that she or he will be charged for
+    *                      the list objects request in V2 style.
+    *                      Bucket owners need not specify this parameter in their requests.
+    * @return an [[Observable]] that emits the [[S3Object]]s.
+    */
+  private def listingHelp[A](
+    bucket: String,
+    sort: (S3Object, S3Object) => Boolean,
+    prefix: Option[String] = None,
+    requestPayer: Option[RequestPayer] = None): Observable[Buffer[S3Object]] = {
+    Observable
+      .fromTask(Task.from {
+        this.s3Client.listObjectsV2(
+          S3RequestBuilder.listObjectsV2(bucket, prefix = prefix, requestPayer = requestPayer))
```

Comment:
we should not use this method since it will not return all the list of keys, by default only `1000` .
there is already implemented a [listObjects](https://github.com/monix/monix-connect/blob/master/s3/src/main/scala/monix/connect/s3/S3.scala#L877) method that would fit very well here, where the sort operation will not be done with `List` and `sortWith` ,but implemented with an `Observable` and an operator like `fold`.

Reply:
I was looking listObjects to do this, but I became confused here with how much it can return. Is there a return everything with listObjects or not? That's why I used the listObjectsV2 I though it returned everything.

Reply:
a single `ListObjectsV2Request` will return at most 1000 objects, and a token with it to list the next batch,
so the `listObjects` already iterates over and will return everything, unless you explicitly set a limit with `maxTotalKeys`  :)

Reply:
I think thats where I got confused, I though if you didn't set maxTotalKeys it defaulted to 1. Thanks for clearing up the confusion.

Reply:
would reduce work?

Reply:
sorry didn't get the question, what do you mean?